choreolib/src/main/native/cpp/choreo/lib/ChoreoTrajectoryState.cpp

```diff
@@ -0,0 +1,92 @@
+#include "choreo/lib/ChoreoTrajectoryState.h"
+
+#include <frc/geometry/Twist2d.h>
+#include <numbers>
+#include <wpi/json.h>
+
+using namespace choreolib;
+
+namespace frc {
+static constexpr frc::Pose2d Interpolate(const frc::Pose2d &startValue,
+		const frc::Pose2d &endValue, double t) {
+	if (t < 0) {
+		return startValue;
+	} else if (t >= 1) {
+		return endValue;
+	} else {
+		frc::Twist2d twist { startValue.Log(endValue) };
+		frc::Twist2d scaledTwist = twist * t;
+		return startValue.Exp(scaledTwist);
+	}
+}
+} // namespace frc
+
+ChoreoTrajectoryState::ChoreoTrajectoryState() : timestamp(0_s), x(0_m), y(0_m), heading(
+		0_rad), velocityX(0_mps), velocityY(0_mps), angularVelocity(0_rad_per_s) {
```

Comment:
Don't initialize the members's default values here. You can assign them directly in the header.